src/Aspire.Dashboard/wwwroot/js/app.js

```diff
@@ -299,3 +299,25 @@ window.listenToWindowResize = function(dotnetHelper) {
 
     window.addEventListener('resize', throttledResizeListener);
 }
+
+window.registerOpenTextVisualizerOnClick = function(layout) {
+    const onClickListener = function (e) {
+        const text = e.target.getAttribute("data-textvisualizer-text");
+        const description = e.target.getAttribute("data-textvisualizer-description");
+
+        if (e.target.tagName.toLowerCase() === "fluent-menu-item" && text && description) {
+            e.stopPropagation();
+            layout.invokeMethodAsync("OpenTextVisualizerAsync", text, description);
+        }
+    }
+
+    document.addEventListener('click', onClickListener);
+
+    return {
+        onClickListener: onClickListener,
+    }
+}
+
+window.unregisterOpenTextVisualizerOnClick = function (listener) {
+    document.removeEventListener('click', listener);
```

Comment:
`registerOpenTextVisualizerOnClick` is returning an object with the function as the `onClickListener` property value.

`unregisterOpenTextVisualizerOnClick` is passed the object from the server, but the `onClickListener` property value isn't accessed to get the func to pass to removeEventListener.

It doesn't look like this would work to me.

Reply:
fixed